lib/go-parser/symbols/index.ts

```diff
@@ -0,0 +1,203 @@
+    if (
+      b.length < 16 ||
+      b[4] !== 0 ||
+      b[5] !== 0 ||
+      // pc quantum
+      (b[6] !== 1 && b[6] !== 2 && b[6] !== 4) ||
+      // pointer size
+      (b[7] !== 4 && b[7] !== 8)
+    ) {
+      throw new Error("unknown header format");
+    }
+
+    // determine the endianness and the PCLN Table version
+    const leMagic = b.readUInt32LE(0);
+    const beMagic = b.readUInt32BE(0);
+    if (leMagic === LineTable.go12magic) {
+      this.binary = littleEndian;
+      this.version = pclnVersion.v12;
+    } else if (beMagic === LineTable.go12magic) {
+      this.binary = bigEndian;
+      this.version = pclnVersion.v12;
+    } else if (leMagic === LineTable.go116magic) {
+      this.binary = littleEndian;
+      this.version = pclnVersion.v116;
+    } else if (beMagic === LineTable.go116magic) {
+      this.binary = bigEndian;
+      this.version = pclnVersion.v116;
+    } else if (leMagic === LineTable.go118magic) {
+      this.binary = littleEndian;
+      this.version = pclnVersion.v118;
+    } else if (beMagic === LineTable.go118magic) {
+      this.binary = bigEndian;
+      this.version = pclnVersion.v118;
+    } else {
+      throw new Error("unknown / unsupported Go version");
+    }
+
+    this.ptrsize = b[7];
+    const uintptr = (b: Buffer): bigint => {
+      if (this.ptrsize === 4) {
+        return this.binary.Uint32(b);
+      }
+      return this.binary.Uint64(b);
+    };
+    const offset = (word: number): bigint => {
+      return uintptr(b.slice(8 + word * this.ptrsize));
+    };
+    const data = (word: number): Buffer => {
+      return b.slice(Number(offset(word))); // TODO: int conversion?
+    };
+
+    switch (this.version) {
+      case pclnVersion.v118:
+        this.nfiletab = Number(offset(1));
+        this.filetab = data(5);
+        this.funcdata = data(7);
+        break;
+      case pclnVersion.v116:
+        this.nfiletab = Number(offset(1));
+        this.filetab = data(4);
+        this.funcdata = data(6);
+        break;
+      case pclnVersion.v12:
+        const nfunctab = Number(this.uintptr(b.slice(8)));
+        this.funcdata = b;
+        const functab = b.slice(8 + this.ptrsize);
+        const functabsize = (nfunctab * 2 + 1) * this.functabFieldSize();
+        const fileoff = this.binary.Uint32(functab.slice(functabsize));
+        this.filetab = b.slice(Number(fileoff));
+        this.nfiletab = Number(this.binary.Uint32(this.filetab));
+        break;
+      default:
+        throw new Error("unreachable");
+    }
+  }
+
+  // go12MapFiles returns a list of files that have been found in the symbol table.
+  // In the original Go implementation, this function takes a map and object, but
+  // we don't need to construct a map and don't need the object.
+  // https://cs.opensource.google/go/go/+/refs/tags/go1.18.5:src/debug/gosym/pclntab.go;l=669
+  public go12MapFiles(): string[] {
+    this.initFileMap();
+    const m: string[] = [];
```

Comment:
I know this is basically a JS translation to a golang code, but this type of variable naming, though common in Go, is not considered best practice in JS. Please try to name variables in a more meaningful way

Reply:
Done 👍(I think 😄)